src/index.html

```diff
@@ -6,41 +6,40 @@
       name="viewport"
       content="width=device-width, initial-scale=1.0"
     />
-    <title>Promise function</title>
-    <link
-      rel="stylesheet"
-      href="./styles/main.scss"
-    />
+    <title>Document</title>
   </head>
   <body>
-    <form>
-      <label for="login">
-        Login
-        <input
-          name="login"
-          type="text"
-          placeholder="Login"
-          id="login"
-        />
-      </label>
-      <label for="password">
-        Password
-        <input
-          name="password"
-          type="password"
-          placeholder="Password"
-          id="password"
-        />
-      </label>
+    <script>
+      function waitFor(element, eventName) {
+        return new Promise((resolve) => {
+          element.addEventListener(eventName, function handler(event) {
+            element.removeEventListener(eventName, handler);
+            resolve(
+              `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+            );
+          });
+        });
+      }
```

Comment:
The `waitFor` function is not completely reliable. It assumes that the event will definitely happen in the future. However, if the event has already happened before this function is called, then the promise will never resolve. To make this function more reliable, you could check if the event has already happened, and resolve immediately if it is.